### PR TITLE
helm charts for decco and other resources

### PR DESCRIPTION
**Pull request by brandune**
Helm charts for decco! This also includes the resources that were appended by kplane-tests, some of which are shared class resources that get their own simple chart since other components besides decco depend on them.

The only changes to the templates are for helm templating, no others were necessary.

I've left the original manifests in so we can refactor the deployment/test processes before cleaning them up and not having to do all of it synchronously.

### CODE REVIEW

File: deploy/decco/templates/decco-apps-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (erwinvaneyk) [inline deploy/decco/templates/decco-apps-crd.yaml:1]: Maybe you could move the crds to the `crds/` directory, which seems to be a best practice in Helm (see https://helm.sh/docs/chart_best_practices/custom_resource_definitions/)

File: deploy/decco/templates/decco-ns.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    decco-project: system
```
Comment (erwinvaneyk) [inline deploy/decco/templates/decco-ns.yaml:5]: Is there a reason/convention why we are using this label?

File: deploy/decco/templates/required.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# This template is used to force the evaluation of vars that are required using the
```
Comment (erwinvaneyk) [inline deploy/decco/templates/required.yaml:1]: Cool! 😄

File: deploy/decco/values.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# wildcard_tls_cert_b64: ''
+# wildcard_tls_key_b64: ''
+
+# tcp_tls_ca_b64: ''
+# tcp_tls_cert_b64: ''
+# tcp_tls_key_b64: ''
+
+
+# This last value is a secret as well, but is actually optional.
+# slack_webhook_for_dns_update_failure: ''
```
Comment (erwinvaneyk) [inline deploy/decco/values.yaml:20]: It is optional, so you could remove the comment to highlight that the default value is `''`.

File: deploy/decco/templates/nginx-ingress-controller.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        image: quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.19.0
```
Comment (erwinvaneyk) [inline deploy/decco/templates/nginx-ingress-controller.yaml:37]: Maybe extract this image as a parameter

File: deploy/decco/templates/default-http-backend.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    matchLabels:
+      k8s-app: default-http-backend
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        k8s-app: default-http-backend
+    spec:
+      containers:
+      - image: gcr.io/google_containers/defaultbackend:1.0
```
Comment (erwinvaneyk) [inline deploy/decco/templates/default-http-backend.yaml:20]: Maybe extract this image as a parameter

File: deploy/decco/templates/tcp-cert.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
Comment (bcle) [inline deploy/decco/templates/tcp-cert.yaml:1]: I have a feeling this secret is not used anywhere and we might be able to delete it.

Comment (brandune) [reply deploy/decco/templates/tcp-cert.yaml:1]: I'm always happy to remove extraneous stuff, but I want to make sure I understand why.

AFAICT, the only App that gets deployed to a Space in the decco Namespace is kplane-usermgr, which has one HTTP endpoint. Still, removing this secret would prohibit TCP endpoints from being added in the future. Is that right?

Comment (bcle) [reply deploy/decco/templates/tcp-cert.yaml:1]: Unless things have changed, AFAIK kplane-usermgr is never deployed into the decco namespace. Normally, a space resource is created in the decco namespace, resulting in a new namespace (the "kplane namespace"), where kplane-usermgr is then deployed. Is my understanding correct?

Comment (bcle) [reply deploy/decco/templates/tcp-cert.yaml:1]: Never mind, I think I'm wrong. Space resources do get created in decco ns, but they need the tcp-cert secret. So ignore what I said. It's all well.

Comment (brandune) [reply deploy/decco/templates/tcp-cert.yaml:1]: 👍

**Pull request merged**